### PR TITLE
[Optimizer2] Implement type conversion from ExecutionNodes -> PlanNodes

### PR DESCRIPTION
**Pull request by hkernbach**
### Scope & Purpose

This PR will introduce a new method for all usable ExecutioNode types.
See code example for basic example:

```
/*
auto DocumentProducingNode::toInspectable() {
  return arangodb::aql::optimizer2::nodes::DocumentProducingNode{
    .count = _readOwnWrites // ... 
  };
}*/
```

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/Aql/ExecutionNode.cpp [hunk 1]
```diff
@@ -2582,6 +2584,51 @@ ReturnNode::ReturnNode(ExecutionPlan* plan,
...
+  }
+
+  // Think about getting rid of this in total.
+  CostEstimate estimate = getCost();
+  // Think about getting rid of this in total.
+  planReturnNode.estimatedCost = estimate.estimatedCost;
+  // Think about getting rid of this in total.
+  planReturnNode.estimatedNrItems = estimate.estimatedNrItems;
+  // "canThrow" attribute seems to be ignored in this case here.
+  // TODO - fixme end
```
Comment (hkernbach) [inline arangod/Aql/ExecutionNode.cpp:2627]: @markuspf important todo. We need to solve this "cleanly". Currently just implemented as it is as a reminder. This affects all "PlanNodes".

File: arangod/Aql/Optimizer2/PlanNodes/ReturnNode.h [hunk 2]
```diff
@@ -26,6 +26,9 @@
 #include "Aql/Optimizer2/PlanNodeTypes/Variable.h"
 #include "Aql/Optimizer2/PlanNodeTypes/Count.h"
 #include "Inspection/VPackInspection.h"
+// TODO: ^-- compiling stops here after I've included ReturnNode into
+// ExecutionNode.cpp with "file not found". Guess, this needs to be fixed in
+// cmake.
```
Comment (hkernbach) [inline arangod/Aql/Optimizer2/PlanNodes/ReturnNode.h:31]: Fails currently with:
```
[ 81%] Building CXX object arangod/Aql/CMakeFiles/arango_aql.dir/IndexHint.cpp.o
In file included from /Users/hkernbach/Git/ArangoDB/devel/arangod/Aql/ExecutionNode.cpp:73:
/Users/hkernbach/Git/ArangoDB/devel/arangod/Aql/Optimizer2/PlanNodes/ReturnNode.h:28:10: fatal error: 'Inspection/VPackInspection.h' file not found
#include "Inspection/VPackInspection.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

we need to fix cmake. up until now, this inspection header file has been used only in our test environment I guess.

Comment (hkernbach) [reply arangod/Aql/Optimizer2/PlanNodes/ReturnNode.h:31]: @markuspf fyi.

File: arangod/Aql/CollectNode.cpp [hunk 3]
```diff
@@ -146,6 +148,58 @@ void CollectNode::doToVelocyPack(VPackBuilder& nodes,
   _options.toVelocyPack(nodes);
 }
 
+optimizer2::nodes::CollectNode CollectNode::toInspectable() const {
+  // CONTINUE HERE TODO HEIKO TODO
```
Comment (hkernbach) [inline arangod/Aql/CollectNode.cpp:152]:
```suggestion
```

**Review by markuspf — APPROVED** [review]

**Pull request merged**